lib/VerticalDelegate.js

```diff
@@ -433,10 +434,10 @@ module.exports = {
 
 		// if we need to refresh, do it now and ensure that we're properly setup to scroll
 		// if we were adding to a partially filled page
-		if (props.index <= end ) this.refresh(list);
+		if (idx <= end ) this.refresh(list);
```

Comment:
If starting index of added models is greater than `end` which is last index of pages, we do not need to generate pages.
But, if it is equal or less than `end`, current pages should be updated.
I think you don't need to subtract 1.